### PR TITLE
Start of ForgeIngameGui

### LINKED ISSUE #9: Required features for Classic Bars
[CurseForge](https://www.curseforge.com/minecraft/mc-mods/classic-bars)

"a nice test for screen render events"

* [x] `patchwork-registries`
  * [x] ForgeRegistries
* [x] `patchwork-fml`
  * [x] ObfuscationReflectionHelper (related issue: https://github.com/PatchworkMC/patchwork-patcher/issues/5)
  * [x] ModList
  * [x] ConfigChangedEvent
* [ ] `patchwork-rendering`
  * [ ] ForgeIngameGui
  * [ ] RenderGameOverlayEvent
* [x] `patchwork-networking`
  * [x] NetworkRegistry
  * [x] SimpleChannel
  * [x] NetworkDirection
  * [x] NetworkEvent.Context
* [x] `patchwork-events-entity`
  * [x] PlayerEvent.PlayerLoggedInEvent
  * [x] LivingEvent.LivingUpdateEvent

### PR DESCRIPTION
**Pull request by TheGlitch76**
I tried to clean this up, but the original Vanilla code is just decompiled
spaghetti. Whoever made this put a lot of effort into making this code
readable.

These changes are so much code I opted to split them up into classes for each
method Forge overwrites/modifies in Vanilla. Hopefully this will keep the line
count down to a reasonable level.

I figure there's no reason why any mod should call the internal code,
so it's `@Unique`. I still preserved the original access levels, however.

Resolves #9

### CODE REVIEW

File: patchwork-gui/src/main/java/net/patchworkmc/mixin/gui/gameoverlay/HookMainEvent.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	@Unique
+	private boolean mainCanceled = false;
+
+	// Technically this hook happens a tiny bit later, but it shouldn't make a difference.
+	@Inject(method = "render", at = @At("HEAD"), cancellable = true)
+	private void preMainEvent(float tickDelta, CallbackInfo info) {
+		PatchworkGui.eventParent = new RenderGameOverlayEvent(tickDelta, this.client.window);
+		ForgeIngameGui.renderHealthMount = client.player.getVehicle() instanceof LivingEntity;
+		ForgeIngameGui.renderFood = client.player.getVehicle() == null;
+		// ForgeIngameGui.renderJumpBar = client.player.isRiding();
```
[the remark this answers is not in the document]
Comment (TheGlitch76) [reply patchwork-gui/src/main/java/net/patchworkmc/mixin/gui/gameoverlay/HookMainEvent.java]: I've commented out flags that don't events yet.

**Comment by TheGlitch76** [comment]
I have no interest in maintaining this right now, if someone wants to try and tackle coderbot's suggestions above they can feel free to push/PR to this branch.

File: patchwork-gui/src/main/resources/patchwork-gui.mixins.json [hunk 2]
```diff
@@ -3,11 +3,13 @@
   "package": "net.patchworkmc.mixin.gui",
   "compatibilityLevel": "JAVA_8",
   "client": [
-    "MixinAbstractInventoryScreen",
-    "MixinGameRenderer",
-    "MixinKeyboard",
-    "MixinMouse",
-    "MixinScreen"
+    "gameoverlay.HookMainEvent",
```
[the remark this answers is not in the document]
Comment (ZNixian) [reply patchwork-gui/src/main/resources/patchwork-gui.mixins.json]: Fixed by #100.

File: patchwork-gui/src/main/java/net/minecraftforge/client/event/RenderGameOverlayEvent.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+		HEALTHMOUNT,
+		// JUMPBAR,
+		// CHAT,
+		// PLAYER_LIST,
+		// DEBUG,
+		// POTION_ICONS,
+		// SUBTITLES,
+		// FPS_GRAPH,
+		// VIGNETTE
+	}
```
[the remark this answers is not in the document]
Comment (kitlith) [reply patchwork-gui/src/main/java/net/minecraftforge/client/event/RenderGameOverlayEvent.java:91]: so far as I can tell, this is/was already above the inner classes? Did you mean something else?

**Comment by TheGlitch76** [comment]
I'm closing this for now. If someone wants to take this back up they can do it in a separate PR.

**Pull request closed without merging**